src/main/java/com/google/devtools/build/remote/client/LogParserUtils.java

```diff
@@ -0,0 +1,155 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.remote.client;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.google.common.collect.Multiset;
+import com.google.common.collect.TreeMultiset;
+import com.google.devtools.build.lib.remote.logging.RemoteExecutionLog.LogEntry;
+import com.google.devtools.build.lib.remote.logging.RemoteExecutionLog.RpcCallDetails.DetailsCase;
+import com.google.devtools.build.lib.remote.logging.RemoteExecutionLog.WatchDetails;
+import com.google.devtools.build.remote.client.RemoteClientOptions.PrintLogCommand;
+import com.google.devtools.remoteexecution.v1test.ExecuteResponse;
+import com.google.longrunning.Operation;
+import com.google.longrunning.Operation.ResultCase;
+import com.google.protobuf.Timestamp;
+import com.google.watcher.v1.Change;
+import com.google.watcher.v1.Change.State;
+import com.google.watcher.v1.ChangeBatch;
+import io.grpc.Status.Code;
+import java.io.BufferedWriter;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Methods for printing log files.
+ */
+public class LogParserUtils {
+
+  private static final String DELIMETER =
+      "---------------------------------------------------------\n";
+
+  public static class ParamException extends Exception {
+
+    public ParamException(String message) {
+      super(message);
+    }
+  }
+
+  private String filename;
+
+  public LogParserUtils(String filename) {
+    this.filename = filename;
```

Comment:
Why do you need a state here anyway? Usually a utils is a collection of static methods, I suspect it would be much simpler here.